src/machines/java/com/enderio/machines/common/recipe/SagMillingRecipe.java

```diff
@@ -282,6 +282,14 @@ public SagMillingRecipe fromJson(ResourceLocation recipeId, JsonObject serialize
                     // Get tag
                     ResourceLocation id = new ResourceLocation(obj.get("tag").getAsString());
                     TagKey<Item> tag = ItemTags.create(id);
+
+                    // TODO: move these tests into OutputItem instead..
```

Comment:
What are these checks added for? I think I can leave these for a future 1.20 todo cleanup pass though.

Reply:
I think that invalid recipes (ingredients without valid items (empty tags, missing mods) etc. ) are not registered and don't pollute the recipe search, but it really shouldn't be in this PR

Reply:
That's all good then, I'll leave it in there and it can get picked up again (one day ™️)